### PR TITLE
Fix dead increments detected by clang CI

### PR DESCRIPTION
**Pull request by ghost**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

...

**Test plan**

...

**Closing issues**

...

### CODE REVIEW

**Review by ret2libc — APPROVED** [review]

**Review by XVilka — APPROVED** [review]

File: librz/core/canalysis.c [hunk 1]
```diff
@@ -1911,7 +1911,7 @@ RZ_API int rz_core_analysis_esil_fcn(RzCore *core, ut64 at, ut64 from, int refty
 		}
 		esil = RZ_STRBUF_SAFEGET (&op->esil);
 		eprintf ("0x%08"PFMT64x" %d %s\n", at, op->size, esil);
-		at += op->size;
+		// at += op->size;
```
Comment (ret2libc) [inline librz/core/canalysis.c:1914]: why commenting it?

File: librz/core/core.c [hunk 2]
```diff
@@ -1685,7 +1685,6 @@ RZ_API void rz_core_autocomplete(RZ_NULLABLE RzCore *core, RzLineCompletion *com
 			if (!len || !strncmp (buf->data + chr, sdbkv_key (kv), minlen)) {
 				char *p = strchr (buf->data + chr, '.');
 				if (p) {
-					j += autocomplete_pfele (core, completion, sdbkv_key (kv), pfx, j, p + 1);
```
Comment (ret2libc) [inline librz/core/core.c:1688]: is this really dead? maybe the assignment is useless, but the call to `autocomplete_pfele` seems relevant to me.

**Review by ret2libc — COMMENTED** [review]
I propose you stop there with changes and we start review this first part to not put too many things together. Is that ok?

File: librz/core/canalysis.c (hunk 1 above)
Comment (ghost) [reply librz/core/canalysis.c:1914]: there is some code commented under. And a todo

File: librz/core/core.c (hunk 2 above)
Comment (ghost) [reply librz/core/core.c:1688]: i will check at lunch

**Comment by ghost** [comment]
rebase or "merge and squash"?

**Comment by ret2libc** [comment]
> rebase or "merge and squash"?

As you prefer. Since commits seem well done, I'm ok to also rebase.

File: librz/core/casm.c [hunk 3]
```diff
@@ -705,16 +705,7 @@ static RzList *rz_core_asm_back_disassemble (RzCore *core, ut64 addr, int len, u
 			last_num_invalid = 0;
 		// disassembly underlap
 		} else if (current_buf_pos + current_instr_len < next_buf_pos) {
-			ut32 purge_results = 0;
```
Comment (ret2libc) [inline librz/core/casm.c:711]: why deleted everything here? I think those functions have side effects that cannot be removed.

File: librz/util/utype.c [hunk 4]
```diff
@@ -578,87 +578,111 @@ RZ_API const char *rz_type_func_args_name(Sdb *TDB, RZ_NONNULL const char *func_
...
+static char *strip_dll_prefix(char *func_name) {
+	char *tmp = strstr(func_name, "dll_");
+	if (tmp) {
+		return tmp + 3;
+	}
+
+	return func_name;
+}
+
+static inline bool is_number(const char *str) {
```
Comment (ret2libc) [inline librz/util/utype.c]: would it be the same if you use `rz_str_is_number`?

**Review by ret2libc — CHANGES_REQUESTED** [review]

**Comment by ghost** [comment]
I add a small refactoring of rz_str_isnumber

**Review by ret2libc — APPROVED** [review]

**Pull request merged**